src/main/java/nl/windesheim/codeparser/patterns/CompositePattern.java

```diff
@@ -0,0 +1,94 @@
+package nl.windesheim.codeparser.patterns;
+
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+
+import java.util.List;
+
+/**
+ * Composite pattern.
+ */
+public class CompositePattern implements IDesignPattern {
+    /**
+     * The component of the composite pattern.
+     */
+    private ClassOrInterfaceDeclaration component;
```

Comment:
Dit moeten ClassOrInterface typen zijn in plaats van ClassOrInterfaceDeclaration. Net zoals de andere pattern classes ook hebben